pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1167,9 +1167,13 @@ protected void handleProducer(final CommandProducer cmdProducer) {
                             CompletableFuture<SchemaVersion> schemaVersionFuture = tryAddSchema(topic, schema);
 
                             schemaVersionFuture.exceptionally(exception -> {
+                                String message = exception.getMessage();
+                                if (exception.getCause() != null) {
+                                    message += (" caused by " + exception.getCause().toString());
```

Comment:
```suggestion
                                    message += (" caused by " + exception.getCause());
```

You don't need to use `toString` here. It is problematic as `exception.getCause` can return `null`.

Reply:
Fixed